src/ble/att/mod.rs

```diff
@@ -220,17 +274,77 @@ impl<A: Attributes> AttributeServer<A> {
     fn process_request(
         &mut self,
         pdu: Pdu,
-        _responder: &mut L2CAPResponder,
+        responder: &mut L2CAPResponder,
     ) -> Result<(), AttError> {
         match pdu.params {
-            AttMsg::ReadByGroup {
+            AttMsg::ReadByGroupReq {
                 handle_range,
-                group_type: _,
+                group_type,
             } => {
-                handle_range.check()?;
-                unimplemented!()
+                let (lower, upper) = {
+                    let range = handle_range.check()?;
+                    (range.start().0, range.end().0)
+                };
+
+                let mut buf = [0u8; 16];
+                let mut writer = ByteWriter::new(&mut buf);
+
+                for att in self.attrs.attributes() {
+                    if att.att_type == group_type && att.handle.0 >= lower && att.handle.0 <= upper
+                    {
+                        let data = ByGroupAttData {
+                            handle: att.handle,
+                            end_group_handle: 0,
+                            value: att.value,
+                        };
+
+                        data.to_bytes(&mut writer).unwrap();
+                    }
+                }
+
+                // If no attributes matched request, return AttributeNotFound error, else send ReadByGroupResponse
+                if writer.space_left() == 16 {
+                    let err = AttError {
+                        code: ErrorCode::AttributeNotFound,
+                        handle: AttHandle::NULL,
+                    };
+
+                    debug!("ATT Error: {:?}", err);
+
+                    Err(err)
+                } else {
+                    let length = 2 + 2 + self.attrs.attributes()[0].value.0.len() as u8;
```

Comment:
I'm assuming this is temporary and will be replaced?

Reply:
Definitely yes, I'll fix it now.